ipie/hamiltonians/generic.py

```diff
@@ -48,7 +48,7 @@ class GenericRealChol(GenericBase):
     Can be created by passing the one and two electron integrals directly.
     """
 
-    def __init__(self, h1e, chol, ecore=0.0, verbose=False):
+    def __init__(self, h1e, chol, ecore=0.0, shmem=False, chol_packed=None, verbose=False):
```

Comment:
My instinct tells me that this is a(n) (existing) design mistake and no logic should occur in the constructor. That is, the hamiltonian class should probably just be specialized for the different cases. However, this can be refactored in the future so shouldn't hold you back.

Reply:
thanks Fionn, good to know